From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Grigoris Pavlakis <grigpavl@ece.auth.gr>
Date: Sat, 28 May 2022 17:21:27 +0300
Subject: [PATCH] Disable forced fullscreen on logical scaling

sdl12-compat forces fullscreen on anything that sets video mode
and uses OpenGL logical scaling, causing rapid flickering and preventing
execution. Not sure if this is an upstream bug or intended behavior,
but disabling fullscreen at this point fixes the flickering.
---
 src/SDL12_compat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/SDL12_compat.c b/src/SDL12_compat.c
index 8d07936f35df5f0874e7f8466e79cb9af0397d93..1915ab718f3fa1d40bd168a119e0f19617ab0fc5 100644
--- a/src/SDL12_compat.c
+++ b/src/SDL12_compat.c
@@ -6167,7 +6167,7 @@ SetVideoModeImpl(int width, int height, int bpp, Uint32 flags12)
                 SDL20_SetWindowFullscreen(VideoWindow20, fullscreen_flags20);
                 SDL20_SetWindowSize(VideoWindow20, width, height);  /* not scaled_width, scaled_height */
                 if (was_fullscreen) {
-                    fullscreen_flags20 |= SDL_WINDOW_FULLSCREEN;
+                    /* fullscreen_flags20 |= SDL_WINDOW_FULLSCREEN; */
                     SDL20_SetWindowFullscreen(VideoWindow20, fullscreen_flags20);
                 }
             }
